### PR TITLE
feature/PATRO23-146 - add password policy

### PR DESCRIPTION
**Pull request by IzabelaPalubicka**
https://tracker.intive.com/jira/browse/PATRO23-146

Must be merged with: [https://github.com/intive/patronage2023-dotnet/pull/33](33)

### CODE REVIEW

File: realm-config.json [hunk 1]
```diff
@@ -0,0 +1,2172 @@
+{
+	"id": "03ad796a-c0d0-4627-bb40-e6ac98b9c7d4",
+	"realm": "Example-Realm",
```
Comment (mateuszgiedzinski) [inline realm-config.json:3]: Same problem with realm name as in Sign up config.

**Review by mateuszgiedzinski — CHANGES_REQUESTED** [review]

**Review by mateuszgiedzinski — APPROVED** [review]
LGTM

File: docker-compose.override.yml [hunk 2]
```diff
@@ -24,7 +24,12 @@ services:
       - ${APPDATA}/ASP.NET/Https:/root/.aspnet/https:ro
       - ${USERPROFILE}/.aspnet/https:/https/
   keycloak:
-    command: start-dev
+    command:
+     - "-Dkeycloak.migration.action=import"
+     - "-Dkeycloak.migration.provider=singleFile"
+     - "-Dkeycloak.migration.file=/root/data/import/realm-config.json"
+     - "-Dkeycloak.migration.strategy=IGNORE_EXISTING"
```
Comment (mateuszmuszynski) [inline docker-compose.override.yml:31]: Can't we just use realm import during startup? https://www.keycloak.org/server/importExport#_importing_a_realm_during_startup
What's the advantage of your approach?

File: src/api/app/Keycloak/Dockerfile [hunk 3]
```diff
@@ -14,6 +14,3 @@ RUN /opt/keycloak/bin/kc.sh build
 
 FROM quay.io/keycloak/keycloak:latest
 COPY --from=builder /opt/keycloak/ /opt/keycloak/
-
-# change these values to point to a running postgres instance
-ENTRYPOINT ["/opt/keycloak/bin/kc.sh"]
```
Comment (mateuszmuszynski) [inline src/api/app/Keycloak/Dockerfile:19]: Why has it been removed?

**Review by mateuszmuszynski — CHANGES_REQUESTED** [review]

File: docker-compose.override.yml (hunk 2 above)
Comment (mateuszmuszynski) [reply docker-compose.override.yml:31]: It's possible to import realm in keyclock user interface. Maybe we can just leave configuration files as it is and import it manually for both dev and prod. What do you think?

Comment (pzajaczkowski) [reply docker-compose.override.yml:31]: Imo not having to import realm every time we create Keycloak container would be nice. At least for dev. Don't really care what import approach is used.

Comment (IzabelaPalubicka) [reply docker-compose.override.yml:31]: I tried this approach from link and it didn't work. Line 31 if file exist ignore.

File: src/api/app/Keycloak/Dockerfile (hunk 3 above)
Comment (IzabelaPalubicka) [reply src/api/app/Keycloak/Dockerfile:19]: Because it doesn't work. In docker-compose.override.yml in keycloak commands we have start dev.

**Comment by mateuszgiedzinski** [comment]
Conflict should be resolved.

**Review by pzajaczkowski — APPROVED** [review]
👍

**Comment by IzabelaPalubicka** [comment]
The story has changed, so I corrected the password requirements and I also changed that there is no username field on registration (only email)

**Pull request merged**